File: bindings/node/examples/crud.js
#!/usr/bin/env node
const fs = require('node:fs')
const os = require('node:os')
const path = require('node:path')

const { Database } = require('../main.js')

function tempDbPath() {
  const dir = fs.mkdtempSync(path.join(os.tmpdir(), 'sombra-example-'))
  return path.join(dir, 'db')
}

async function main() {
  const db = Database.open(tempDbPath()).seedDemo()

  const newUserId = db.createNode('User', { name: 'Example User', bio: 'Hello from Node' })
  console.log('Created user id:', newUserId)

  db.updateNode(newUserId, { set: { bio: 'Updated bio' } })

  const rows = await db
    .query()
    .match('User')
    .where('n0', (pred) => pred.eq('name', 'Example User'))
    .select(['n0'])
    .execute()
  console.log('Query results:', rows)

  db.deleteNode(newUserId, true)
  console.log('Deleted user', newUserId)
}

main()

File: bindings/python/examples/crud.py
"""Simple CRUD walkthrough for the Python bindings."""

from __future__ import annotations

import tempfile
from pathlib import Path

from sombra import Database
from sombra.query import eq


def temp_db_path() -> str:
    directory = Path(tempfile.mkdtemp())
    return str(directory / "db")


def main() -> None:
    db = Database.open(temp_db_path())
    db.seed_demo()

    user_id = db.create_node("User", {"name": "Example User", "bio": "hello"})
    print("Created user:", user_id)

    db.update_node(user_id, set_props={"bio": "Updated from Python"})

    rows = db.query().nodes("User").where(eq("name", "Example User")).execute()
    print("Query rows:", rows)

    db.delete_node(user_id, cascade=True)
    print("Deleted user:", user_id)


if __name__ == "__main__":
    main()
